hs/src/Reach/Connector/ALGO.hs

```diff
@@ -3015,9 +3033,30 @@ cmeth sigi = \case
 
 bindFromArgs :: [DLVarLet] -> App a -> App a
 bindFromArgs vs m = do
-  -- XXX deal with >15 args
-  let go (v, i) = sallocVarLet v False (code "txna" ["ApplicationArgs", texty i] >> cfrombs (varLetType v))
-  foldl' (flip go) m (zip vs [(1::Integer) ..])
+  let goSingle (v, i) = sallocVarLet v False (code "txna" ["ApplicationArgs", texty i] >> cfrombs (varLetType v))
+  let goSingles singles k =
+        foldl' (flip goSingle) k (zip singles [(1 :: Integer) ..])
+  -- When there are more than 15 args, the 15th arg is a tuple of args 15+
+  case 15 < (length vs) of
+    False -> do
+      goSingles vs m
+    True -> do
+      -- TODO - I wrote this thinking it was the code path I needed to edit, but it's not exercised by my test.  I should write a test that exercises it and be sure it works.
```

Comment:
This code is used by View functions, so you'd need to make a View function with 15+ args and call it

Reply:
I think I don't understand views.  I thought that a view was basically computed client-side after observing the contract state, which doesn't require a publication to request anything of the contract or cause any computation in the contract.  That was what I understood when working on the stdlib code for views.  But this looks like views are more like APIs, that you can publish to the contract to compute something in the consensus state.  So... how do views work?

Reply:
When you write a View, it is part of the published API and it can be called on-chain.

ALSO, Reach knows how to decode the state and run it locally as JS.

This is because on ALGO, there are no free/read-only functions like there are on EVM, so we had to implement them ourselves. Once ALGO supports them, we'll delete the existing client-side view implementation code and just call the node itself.